app/src/main/java/com/google/samples/apps/sunflower/compose/PlantDetailView.kt

```diff
@@ -0,0 +1,505 @@
+import androidx.ui.foundation.VerticalScroller
+import androidx.ui.foundation.drawBackground
+import androidx.ui.foundation.shape.corner.CircleShape
+import androidx.ui.foundation.shape.corner.RoundedCornerShape
+import androidx.ui.graphics.Color
+import androidx.ui.layout.Arrangement
+import androidx.ui.layout.Column
+import androidx.ui.layout.ColumnScope.gravity
+import androidx.ui.layout.Row
+import androidx.ui.layout.Spacer
+import androidx.ui.layout.Stack
+import androidx.ui.layout.fillMaxSize
+import androidx.ui.layout.fillMaxWidth
+import androidx.ui.layout.offset
+import androidx.ui.layout.padding
+import androidx.ui.layout.preferredHeight
+import androidx.ui.layout.size
+import androidx.ui.layout.sizeIn
+import androidx.ui.layout.wrapContentSize
+import androidx.ui.livedata.observeAsState
+import androidx.ui.material.FloatingActionButton
+import androidx.ui.material.IconButton
+import androidx.ui.material.MaterialTheme
+import androidx.ui.material.Surface
+import androidx.ui.material.TopAppBar
+import androidx.ui.material.icons.Icons
+import androidx.ui.material.icons.filled.Add
+import androidx.ui.material.icons.filled.ArrowBack
+import androidx.ui.material.icons.filled.Share
+import androidx.ui.res.stringResource
+import androidx.ui.semantics.accessibilityLabel
+import androidx.ui.text.font.FontWeight
+import androidx.ui.tooling.preview.Preview
+import androidx.ui.unit.Dp
+import androidx.ui.unit.IntSize
+import androidx.ui.unit.dp
+import androidx.ui.viewinterop.AndroidView
+import androidx.ui.viewmodel.viewModel
+import com.google.android.material.snackbar.Snackbar
+import com.google.samples.apps.sunflower.R
+import com.google.samples.apps.sunflower.data.Plant
+import com.google.samples.apps.sunflower.utilities.InjectorUtils
+import com.google.samples.apps.sunflower.viewmodels.PlantDetailViewModel
+import dev.chrisbanes.accompanist.coil.CoilImageWithCrossfade
+import dev.chrisbanes.accompanist.mdctheme.MaterialThemeFromMdcTheme
+
+private const val ParallaxDelta = 2f
+private const val HeaderTransitionOffset = 150f
+private val SunflowerFabShape =
+    RoundedCornerShape(topLeft = 0.dp, topRight = 30.dp, bottomRight = 0.dp, bottomLeft = 30.dp)
+
+/**
+ * As these callbacks are passed in through multiple Composables, to avoid having to name
+ * parameters to not mix them up, they're aggregated in this class.
+ */
+data class PlantDetailsCallbacks(
+    val onFabClicked: () -> Unit,
+    val onBackClicked: () -> Unit,
+    val onShareClicked: () -> Unit
+)
+
+@Composable
+fun PlantDetailsScreen(
+    plantId: String,
+    onBackClicked: () -> Unit,
+    onShareClicked: (String) -> Unit
+) {
+    // ViewModel and LiveDatas needed to populate the plant details info on the screen
+    val plantDetailsViewModel: PlantDetailViewModel = viewModel(
+        factory = InjectorUtils.providePlantDetailViewModelFactory(ContextAmbient.current, plantId)
+    )
+    val plant = plantDetailsViewModel.plant.observeAsState().value
```

Comment:
What about using delegate syntax here?
```suggestion
    val plant by plantDetailsViewModel.plant.observeAsState()
```

Reply:
I prefer to use `.value` as this handles nullability better. Otherwise, apart from the null checks, I have to use the bang-bang operator when calling the `PlantDetails` function

Reply:
ack